internal/cloud/realm/auth.go

```diff
@@ -12,35 +12,82 @@ type loginPayload struct {
 	PrivateAPIKey string `json:"apiKey"`
 }
 
-// AuthResponse is the Realm login response
-type AuthResponse struct {
+// Session is the Realm session
+type Session struct {
 	AccessToken  string `json:"access_token"`
 	RefreshToken string `json:"refresh_token"`
 }
 
-func (c *client) Authenticate(publicAPIKey, privateAPIKey string) (AuthResponse, error) {
-	path := adminAPI + "/auth/providers/mongodb-cloud/login"
+// AuthProfile contains all of the roles associated with a user
+type AuthProfile struct {
+	Roles []Role `json:"roles"`
+}
+
+// Role contains a GrouID field that maps to a project ID
+type Role struct {
+	GroupID string `json:"group_id"`
+}
+
+// AllGroupIDs returns all available group ids for a given user
+func (pd AuthProfile) AllGroupIDs() []string {
+	var arr []string
+	set := map[string]struct{}{}
+	for _, role := range pd.Roles {
```

Comment:
[nit] add a newline here above the `for` (or remove the newline after the `for` loop)